flow/src/types.rs

```diff
@@ -21,20 +21,18 @@ impl Default for FlowProtocol {
 #[derive(Encode, Decode, Clone, PartialEq, Eq, PartialOrd, Ord, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(Debug))]
 #[repr(u8)]
-pub enum FlowState {
-	Init = 0,
-	Active = 1,
-	Paused = 2,
-	Finalizing = 3,
-	Reverting = 4,
```

Comment:
what if you get an issue while reverting?

Reply:
If you are asking from the perspective of having panic during `on_initialize` - the whole chain will stop working.
If the question is about error handling in general, there are no expected issues (unreserve balance operation looks reliable enough). Not sure that I got the question.

Reply:
in case A there needs to be a way to recover. without that state recovering will be hard as you have to check all the balances, blocks and revert respecitvely.

Reply:
Is something wrong with the changes in the current PR?
Is something wrong with the design in general?

Currently, `CampaignFinalizationQueue` holds information about what was scheduled and what was done. It doesn't differ too much from how it was implemented before. It's even easier to debug. No need to use indexes. Instead, there is a separate structure created only for queue handling.
@2075